src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -9999,7 +9999,18 @@ def describe(
         # max            NaN      3.0    NaN
         sorted_percentiles = sorted(percentiles)
         dtypes = self.dtypes
-        query_compiler = self
+        # If we operate on the original frame's labels, then if two columns have the same name but
+        # different dtypes, the JOIN behavior of SnowflakeQueryCompiler.concat will produce incorrect
+        # results. For example, if a frame has two columns named "a", one with `object` dtype and the
+        # other `int64`, the `object` column would have a query compiler for the computed `top`/`freq`,
+        # while the numeric column would have a query compiler for `std` and other numeric aggregations.
```

Comment:
hmm, is that because the other non-numeric columns got dropped during std ? if you have two df with both labels [a, a, c], the concat should be able to take care of it correctly.

Reply:
Yes, the numeric aggregations will drop non-numeric columns, so previously if you had labels `["a", "a", "c"]` but both "a" columns were numeric, there were issues. There were only problems when there was a mix of object and numeric aggregations, since we had to drop columns with inappropriate dtypes before performing the aggregations.

I've updated the comment example to be more concrete.